contracts/crowdsale/validation/TimedCrowdsale.sol

```diff
@@ -0,0 +1,28 @@
+pragma solidity ^0.4.18;
+
+import "../../math/SafeMath.sol";
+import "../Crowdsale.sol";
+
+contract TimedCrowdsale is Crowdsale {
+  using SafeMath for uint256;
+
+  uint256 public startTime;
+  uint256 public endTime;
+
+  function TimedCrowdsale(uint256 _startTime, uint256 _endTime) public {
+    require(_startTime >= now);
+    require(_endTime >= _startTime);
+
+    startTime = _startTime;
+    endTime = _endTime;
+  }
+
+  function hasExpired() public view returns (bool) {
+    return now > endTime;
+  }
+
+  function _preValidatePurchase(address _beneficiary, uint256 _weiAmount) internal {
+    super._preValidatePurchase(_beneficiary, _weiAmount);
+    require(now >= startTime && now <= endTime);
```

Comment:
Consider using 
`require(!hasExpired());`

Reply:
You mean substitute
`require(now >= startTime && now <= endTime);`
for:
`require(now >= startTime);`
`require(!hasExpired());`
?
We still need to check we are after `startTime`, so I'm not sure what the advantage of this would be...

Reply:
Agreed. My bad =P

Reply:
I would abstract that require into another `inTimeRange` modifier, to make the code easier to read and because it may come handy when working with a `TimedCrowdsale`. Thoughts?

Reply:
maybe `withinBuyingPeriod`